docs/tutorial.md

```diff
@@ -737,3 +731,90 @@ Note that even though adding type annotations to .lua files makes it invalid
 Lua, you can still do so and load them from the Lua VM once the Teal package
 loader is installed by calling tl.loader().
 
+
+### Type definitions for third party libraries
+
+You can also create declaration files to annotate the types of third-party Lua
+modules, including C Lua modules. To do so, create a file with the .d.tl
+extension and require it as normal, i.e. `require("minetest")`.
+
+Types defined in this module will will be used as a source of type information 
+checking with `tl check`, even though the real Lua module will be loaded
+instead when requiring the module from Lua or `tl run`.
+
+If the third party library is "globally available", i.e. you do not explicitly
+`require` it, you can inject the types without affecting your generated code
+with a `tlconfig.lua`:
+
+```lua
+return {
+    preload_modules = {
+        "minetest" -- Execute the equivalent of require('minetest') before type-checking the tl script(s)
+    }
+}
+```
+
+#### Visibility
+
+It should be noted that local references in type definition files (or any files
+containing types for that matter) are not available in the requiring module.
+
+This means that constructs like:
+
+```lua
+local MyPointType = record
+    x :number
+    y :number
+end
+
+local MyCompositeType = record
+    center: MyPointType
+end
+
+return {
+    MyPointType = MyPointType,
+    MyCompositeType = MyCompositeType,
+}
+```
+
+will give errors when accessing `MyCompositeType::center`, as `MyPointType` is not
+visible.
```

Comment:
^ Are you sure you're running the latest tl? This section is not correct as of tl 0.6.0 and `master`! The good news is that above example actually works, you can use it like this:

```
local mypoint = require("mypoint")

local t: mypoint.MyCompositeType = {}

t.center.x = 2

local p: mypoint.MyPointType = { x = 2, y = 3 }

t.center = p
```

Reply:
Huh, moves fast, great. If I don't have to re-do argparse and compat53 I should be able to update pretty quickly. (luarocks on windows with vs2019 is... unpleasant).